src/posix/platform/logging.cpp

```diff
@@ -35,7 +35,8 @@
 
 #include <openthread/platform/logging.h>
 
-#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED
+#if (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED) || \
```

Comment:
Why define `otPlatLog` in platform when `LOG_OUTPUT == APP` ?

Reply:
Under `posix` build the posix platform itself is acting as the app. If we don't add it the under `LOG_OUTPUT_APP` we would get an error (missing definition of `otPlatLog`). I guess such a config may have not ben covered by any CI build before.

Reply:
> Under `posix` build the posix platform itself is acting as the app

I think this is not totally true. Should we keep the boundary of posix platform and posix app more clear?

> If we don't add it the under LOG_OUTPUT_APP we would get an error (missing definition of otPlatLog). 

We should fix in a more suitable way?

Thoughts? @bukepo

Reply:
The platform should only define otPlatLog when `LOG_OUTPUT == PLATFORM_DEFINED`. @abtink may I know how to reproduce the compile error you encountered?

Reply:
This PR is intended to add back to our github action CI some TREL posix builds (somewhat addressing the issue https://github.com/openthread/openthread/issues/6472 caused due to removal of `ot-ncp` and previous TREL posix build checks and tests). 

The use of `LOG_OUTPUT_APP` with CLI under posix seems to have been broken (I guess we did not detect it because  this configuration was probably not being built/checked before). This config used to work correctly with NCP posix build (but as you know the `ot-ncp` was removed, so now we want to use CLI based builds).

I want this PR to be merged to somewhat address the lack any of build check for posix TREL build config. It does not make sense to deal with issues related to posix (I think they should be addressed in other PRs). The goal of this PR is to unblock other PRs adding changes in TREL like https://github.com/openthread/openthread/pull/6515 (so that at minimum we have a CI build check for TREL posix).

So in the new pushed commit in this PR I changed the `toranj` config to use `PLATFORM_DEFINED` for the posix builds, so we don't cover or use this config.

Note that use of `LOG_OUTPUT_APP` config is still broken on posix build. @bukepo , to reproduce simply set the `LOG_OUTPUT` to `LOG_OUTPUT_APP` and build posix CLI executable. 
- The reason for this seems to be that in the CLI, the `otPlatLog` is defined `examples/apps/cli/main.c` but in `posix` build does not seem to include this file (`posix` has its own `main.c` and the only place under posix where `otPlatLog` is defined is in `src/posix/platform/logging.cpp`).
- I leave it to you guys to address this (feel free to later change back the `toranj` config to use `OUTPUT_APP` under posix to add build coverage for this config).

Reply:
Thanks for the explanation. The `LOG_OUTPUT_APP` was intentionally not implemented in posix ot-cli because we can get the same effect with `-v` option(see https://github.com/openthread/openthread/blob/main/src/posix/main.c#L367). The simulation platform is for testing purpose, so we should support as many configurations as possible on that platform. And in my opinion, we may just support what we need on POSIX platform. Thoughts? @abtink

Reply:
Can we define a weak `otPlatLog` in posix platform when `LOG_OUTPUT=LOG_OUTPUT_APP` ?